mifosng-provider/src/main/java/org/mifosplatform/portfolio/loanaccount/domain/transactionprocessor/impl/RBILoanRepaymentScheduleTransactionProcessor.java

```diff
@@ -97,13 +97,15 @@ protected Money handleTransactionThatIsALateRepaymentOfInstallment(final LoanRep
                     loanTransaction.getTransactionDate(), installments);
 
             for (final LoanRepaymentScheduleInstallment installment : installments) {
-                if (installment.isInterestDue(currency)
+                if ((installment.isInterestDue(currency)
+                        || installment.getFeeChargesCharged(currency).minus(installment.getFeeChargesPaid(currency)).isGreaterThanZero() || installment
```

Comment:
Use getFeeChargesOutstanding to check any charges pending in installment. Outstanding calculation should include waived amount also